spring-integration-ws/src/main/java/org/springframework/integration/ws/AbstractWebServiceOutboundGateway.java

```diff
@@ -84,15 +87,17 @@ public abstract class AbstractWebServiceOutboundGateway extends AbstractReplyPro
 
 	private boolean webServiceTemplateExplicitlySet;
 
-	public AbstractWebServiceOutboundGateway(final String uri, WebServiceMessageFactory messageFactory) {
+	@SuppressWarnings("NullAway")
+	public AbstractWebServiceOutboundGateway(final String uri, @Nullable WebServiceMessageFactory messageFactory) {
```

Comment:
This is not correct. 
The `WebServiceTemplate` expects from us a `WebServiceMessageFactory` as a not null.
If you see somewhere it is, let's mitigate it over there!

Reply:
then.. when ctors pass null for `WebServiceMessageFactory`, I'll replace null with `SaajSoapMessageFactory` as the default. Does this sound good?

Based on the [WebServiceTemplate.properties](https://github.com/spring-projects/spring-ws/blob/main/spring-ws-core/src/main/resources/org/springframework/ws/client/core/WebServiceTemplate.properties), `SaajSoapMessageFactory` is indeed the default strategy.

Reply:
Why would we allow to pass null at all?
The `WebServiceTemplate` does not expect that from us, so let's comeback to not-null for this ctors!

Reply:
Right, I meant that I'll remove `@Nullable` from the constructor parameters and fix the calling sites to pass `SaajSoapMessageFactory` instance instead of null.

For example:
```java
// Instead of
super(uri, null);

// I'll change to  
super(uri, new SaajSoapMessageFactory());

Am I still misunderstanding something?

Reply:
I see:
```
	public MarshallingWebServiceOutboundGateway(DestinationProvider destinationProvider,
			WebServiceTemplate webServiceTemplate) {
		super(destinationProvider, null);
		doSetWebServiceTemplate(webServiceTemplate);
	}
```
I think it is better to introduce a `protected` ctor in the `AbstractWebServiceOutboundGateway` based on the `WebServiceTemplate` then.